### ISSUE TITLE
dirPagination: Pagination not working after switch from asynchronous table to synchronous

### ISSUE THREAD

**Issue by ArkoThor** [michaelbromley/angularUtils#174]
Hi, I'm using your pagination directive in project with combination of multiple synchronous and asynchronous tables on different pages and not using pagination-ids. It works great on all of them, only problem is when user switches from page with asynchronous table to page with synchronous one. Then only asynchronous work without refreshing the application.
In function dirPaginationLinkFn in dirPagination.js there is
```javascript
if (typeof attrs.totalItems !== 'undefined') {
                    paginationService.setAsyncModeTrue(paginationId);
```
which switches on async mode, but then in else branch it doesn't switch the mode off.
I have changed it for my project to also switch off async mode by
```javascript
} else {
                    paginationService.setAsyncModeFalse(paginationId);
```
and the switching from synchronous to asynchronous tables and back does not break paging for synchronous tables now. If it doesn't break anything else, it would probably be nice to fix it also in master. 

And thanks for great directive :)

**Comment by michaelbromley** [comment]
Thanks for the report. Looks like you might have found a bug there. I'll look into it and fix as appropriate when I work on the next release.

**Comment by adamcee** [comment]
Hi! I'm looking into using the dirPagination directive for our organization and like what I see so far :). Related to this, I was wondering if there is a timeline/estimate for the next release.

**Comment by michaelbromley** [comment]
Hi @adamcee . There is no timeline I'm afraid. The way it works is this: When I get a bit of free time, I take a look over the issues and if I think I have time to do something with one of them, and if it seems to be a popular request, I try to fix it. Usually I can, but if it is especially complex it may take a few tries.

I've not had a go at this particular issue yet. Is it blocking your current project?

**Comment by adamcee** [comment]
Hi Michael,

Thanks - appreciate your answer & swiftness! We are specifically interested
in using the pagination directive but I don't anticipate this issue being a
blocker. Since it was a bug related to the feature we're interested in
though I thought I'd ask.

**Comment by ArkoThor** [comment]
@adamcee I can only recommend using this directive. We had no other issue with it on our project since this bug. And this one can be solved really simply with the little change I wrote previously. Sure it would be nicer to have it working out of box, but you can always just fix it yourself for your project this way if it causes the same problem to you.

**Comment by adamcee** [comment]
@ArkoThor -- thank you! I appreciate the recommendation very much. I hope
to have it implemented and working in our application (which is still
pretty small) today! Frankly, my only concerns were if it was overkill for
our project (this is my first Angular project so I'm still getting
acclimated), though we will be doing both synchronous and asynchronous
paging -- but in different places -- and also the general concern of
becoming reliant on a 3rd-party dependency. But it's a pretty small
dependency ;) and this is also, imho, a good chance for us to test out
using this tool and then possibly use it in future Angular projects.
@michaelbromley, out of curiosity, have you considered reaching out to the
ui-bootstrap or angularstrap communities (again I'm thinking of project
dependency / contributor size questions) to see if they are interested in
bringing this code into their fold? I'm not sure if it's precisely in line
with what they're hoping to accomplish, but it seems like a great tool....

**Comment by michaelbromley** [comment]
@adamcee No, I've not. I only happen to use the Bootstrap markup since it's probably the most sensible default. Other than that, there's not much overlap in our aims. Also I'm not sure if there is an advantage for either party in doing so. Perhaps more exposure for my module, but in any case I think it's pretty visible if you are searching for "angularjs pagination".

**Comment by matheusdev** [comment]
@michaelbromley I converted the solution proposed by @ArkoThor into a PR. The tests are still passing so this correction will not break other things.

For who is using asynchronous and synchronous tables one quick fix is to give a `pagination-id` for each table, this way every table will not have any relationship with each other.

**Comment by michaelbromley** [comment]
Thanks for the help on this - this fix is now available in v0.10.1 :+1:
